astropy/table/table.py

```diff
@@ -1244,18 +1245,30 @@ def __setitem__(self, item, value):
             name = item
             # If this is a column-like object that could be added directly to table
             if isinstance(value, BaseColumn) or self._add_as_mixin_column(value):
+                # If we're setting a new column to a scalar, broadcast it.
```

Comment:
I can see that that would work well for `Quantity` and `ndarray`, but not so easily for `SkyCoord` and `Time`.

As an alternative, `info` could gain a `broadcast_to` method (or `col_copy`?)